### PR TITLE
enh(apiv2): add extra fields to host read endpoint

### PR DESCRIPTION
**Pull request by victorvassilev**
Resolves: MON-4943

## Description

adds two new fields to the host read api

**Fixes** # MON-4943

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [ ] 19.10.x
- [x] 20.04.x (master)

<h2> How this pull request can be tested ? </h2>

access the api url via GET http://HOST/centreon/api/beta/monitoring/hosts/14/

#### Community contributors & Centreon team

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [x] I have made sure that the **unit tests** related to the story are successful.
- [x] I have made sure that **unit tests cover 80%** of the code written for the story.
- [x] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: src/Centreon/Domain/Monitoring/Host.php [hunk 1]
```diff
@@ -1165,4 +1175,36 @@ public function setCriticality(?int $criticality): Host
...
+        return $this->flapping;
+    }
+
+    /**
+     * @param bool|null $flapping
+     */
+    public function setFlapping(?bool $flapping): void
+    {
+        $this->flapping = $flapping;
+    }
```
Comment (kduret) [inline src/Centreon/Domain/Monitoring/Host.php]:
```suggestion
    /**
     * @param bool|null $flapping
     * @return Host
     */
    public function setFlapping(?bool $flapping): Host
    {
        $this->flapping = $flapping;
        return $this;
    }
```

File: src/Centreon/Domain/Monitoring/Host.php [hunk 2]
```diff
@@ -1165,4 +1175,36 @@ public function setCriticality(?int $criticality): Host
...
+        return $this->percentStateChange;
+    }
+
+    /**
+     * @param float|null $percentStateChange
+     */
+    public function setPercentStateChange(?float $percentStateChange): void
+    {
+        $this->percentStateChange = $percentStateChange;
+    }
```
Comment (kduret) [inline src/Centreon/Domain/Monitoring/Host.php]:
```suggestion
    /**
     * @param float|null $percentStateChange
     * @return Host
     */
    public function setPercentStateChange(?float $percentStateChange): Host
    {
        $this->percentStateChange = $percentStateChange;
        return $this;
    }
```

**Comment by vhr** [comment]
just change the base branch from `master` to `MON-4789-unified-view`

**Comment by victorvassilev** [comment]
> just change the base branch from `master` to `MON-4789-unified-view`

actually I want to merge this into master directly, as it works fine on its own and I don't want to burden the feature branch with extra code

**Pull request merged**